File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()) return false;
        unordered_map<char,int> smap;
        int count=0;
        for(auto it:s){
            smap[it]++;
            count++;
        }
        for(auto it:t){
            if(smap.find(it)!=smap.end() && count!=0){
                if(smap[it]>0){
                    smap[it]--;
                    count--;
                }
            }
        }
        return (count==0);  
    }
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character,Integer> map = new HashMap<>();
        for(char x : s.toCharArray()){
            map.put(x,map.getOrDefault(x,0)+1);
        }

        for(char x : t.toCharArray()){
            int i = map.getOrDefault(x,0);
            if(i==0){return false;}
            map.put(x,i-1);
        }
        int c=0;
        for(int x : map.values()){
            c+=x;
        }

        return (c==0);
    }
}